File: 2772-apply-operations-to-make-all-array-elements-equal-to-zero/2772-apply-operations-to-make-all-array-elements-equal-to-zero.py
class Solution:
    def checkArray(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        for i in range(n-k):
            nums[i+k] += nums[i]
        for i in range(n-1):
            if nums[i]>nums[i+1]:
                return False
            if i>= n-k and nums[i] != nums[i+1]:
                return False
        return True
        

File: 2772-apply-operations-to-make-all-array-elements-equal-to-zero/2772-apply-operations-to-make-all-array-elements-equal-to-zero.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Boolean}
def check_array(nums, k)
  n = nums.size
  for i in 0...(n-k)
    nums[i+k] += nums[i]
  end

  for i in 0...(n-1)
    if nums[i] > nums[i+1]
      return false 
    elsif i>=(n-k) && nums[i] != nums[i+1]
        return false
    end
  end
  return true
end



